sdk/src/wallet/core/operations/stronghold_backup/mod.rs

```diff
@@ -98,212 +96,65 @@ impl Wallet {
             .password(stronghold_password.clone())
             .build(backup_path.clone())?;
 
-        let (read_client_options, read_secret_manager, read_wallet_data) =
-            read_wallet_data_from_stronghold_snapshot::<SecretManager>(&new_stronghold).await?;
+        let (loaded_client_options, loaded_secret_manager_config, loaded_wallet_data, loaded_secret_data) =
+            read_wallet_data_from_stronghold_snapshot::<S>(&new_stronghold).await?;
 
-        let read_bip_path = read_wallet_data.as_ref().and_then(|data| data.bip_path);
+        let loaded_pub_key_opts = loaded_secret_data.as_ref().map(|data| &data.public_key_options);
 
         // If the bip path is not matching the current one, we may ignore the backup
         let ignore_backup_values = ignore_if_bip_path_mismatch.map_or(false, |ignore| {
             if ignore {
                 // TODO: #1279 okay that if both are none we always load the backup values?
-                curr_bip_path != read_bip_path
+                loaded_pub_key_opts.is_some_and(|opts| self.public_key_options() != opts)
             } else {
                 false
             }
         });
 
         if !ignore_backup_values {
-            self.data_mut().await.bip_path = read_bip_path;
+            if let Some(opts) = loaded_pub_key_opts {
+                // TODO
+                // self.secret_data.public_key_options = opts.clone();
```

Comment:
Want to handle in this PR or open an issue?

Reply:
well the problem is that this field is not mutable, and I'm not sure whether it should be

Reply:
It needs to be, unless we change the restore backup in such a way that it creates a new wallet instead of updating the existing one (not sure if that makes sense, but definitely no in this PR)

Reply:
tbh that may be better than dealing with even more lock contention

Reply:
I did it initially as part of the wallet builder and then changed in this commit https://github.com/iotaledger/wallet.rs/pull/1019/commits/037e5c3241a7e267c1e84d317a6681eef439a908, so the restore method can be called on an existing wallet instead of having a different builder (if I remember correctly, Firefly team wanted this change, because it felt a bit weird to them to have the method not available on the wallet, but I think it's not that bad)